src/View/Helper/PaginatorHelper.php

```diff
@@ -1264,10 +1264,11 @@ public function limitControl(array $limits = [], $default = null, array $options
             ];
         }
 
+        $query = $this->_View->getRequest()->getQueryParams();
         $out .= $this->Form->control('limit', $options + [
                 'type' => 'select',
                 'label' => __('View'),
-                'value' => $default,
+                'value' => !empty($query['limit'])?$query['limit']:$default,
```

Comment:
Instead of this conditional we can just use the `default` key to specify default value and this line can be changed to:

```php
'value' => $this->_View->getRequest()->getQuery('limit'),
```